pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -735,3 +747,60 @@ func (s *SlirpPodInterface) setCachedInterface(name string) error {
 	err := writeToCachedFile(&s.domain.Spec.QEMUCmd.QEMUArg[s.podInterfaceNum], qemuArgCacheFile, name)
 	return err
 }
+
+type MacvtapPodInterface struct {
+	vmi              *v1.VirtualMachineInstance
+	vif              *VIF
+	iface            *v1.Interface
+	domain           *api.Domain
+	podInterfaceNum  int
+	podInterfaceName string
+}
+
+func (m *MacvtapPodInterface) discoverPodNetworkInterface() error {
+	if len(m.vif.MAC) == 0 {
+		// Get interface MAC address
+		mac, err := Handler.GetMacDetails(m.podInterfaceName)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to get MAC for %s", m.podInterfaceName)
+			return err
+		}
+		m.vif.MAC = mac
+	}
+
+	return nil
+}
+
+func (m *MacvtapPodInterface) preparePodNetworkInterfaces() error {
+	return nil
+}
+
+func (m *MacvtapPodInterface) decorateConfig() error {
+	m.domain.Spec.Devices.Interfaces[m.podInterfaceNum].MAC = &api.MAC{MAC: m.vif.MAC.String()}
+	m.domain.Spec.Devices.Interfaces[m.podInterfaceNum].Target = &api.InterfaceTarget{
+		Device:  m.podInterfaceName,
+		Managed: "no",
+	}
+	return nil
+}
+
+func (m *MacvtapPodInterface) loadCachedInterface(name string) (bool, error) {
+	var ifaceConfig api.Interface
+
+	isExist, err := readFromCachedFile(name, interfaceCacheFile, &ifaceConfig)
+	if err != nil {
+		return false, err
+	}
+
+	if isExist {
+		m.domain.Spec.Devices.Interfaces[m.podInterfaceNum] = ifaceConfig
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func (m *MacvtapPodInterface) setCachedInterface(name string) error {
```

Comment:
Could you explain why do we need to store these values?

Reply:
I sincerely do not know; just followed the current status quo on the other iface types.